FirebaseFunctions/Sources/Internal/FunctionsSerializer.swift

```diff
@@ -106,73 +91,70 @@ final class FunctionsSerializer: Sendable {
     String(describing: type(of: value))
   }
 
-  private func encodeNumber(_ number: NSNumber) throws -> AnyObject {
-    // Recover the underlying type of the number, using the method described here:
-    // http://stackoverflow.com/questions/2518761/get-type-of-nsnumber
-    let cType = number.objCType
-
-    // Type Encoding values taken from
-    // https://developer.apple.com/library/mac/documentation/Cocoa/Conceptual/ObjCRuntimeGuide/
-    // Articles/ocrtTypeEncodings.html
-    switch cType[0] {
-    case CChar("q".utf8.first!):
-      // "long long" might be larger than JS supports, so make it a string.
-      return ["@type": Constants.longType, "value": "\(number)"] as AnyObject
-
-    case CChar("Q".utf8.first!):
-      // "unsigned long long" might be larger than JS supports, so make it a string.
-      return ["@type": Constants.unsignedLongType,
-              "value": "\(number)"] as AnyObject
-
-    case CChar("i".utf8.first!),
-         CChar("s".utf8.first!),
-         CChar("l".utf8.first!),
-         CChar("I".utf8.first!),
-         CChar("S".utf8.first!):
-      // If it"s an integer that isn"t too long, so just use the number.
-      return number
-
-    case CChar("f".utf8.first!), CChar("d".utf8.first!):
-      // It"s a float/double that"s not too large.
-      return number
-
-    case CChar("B".utf8.first!), CChar("c".utf8.first!), CChar("C".utf8.first!):
-      // Boolean values are weird.
-      //
-      // On arm64, objCType of a BOOL-valued NSNumber will be "c", even though @encode(BOOL)
-      // returns "B". "c" is the same as @encode(signed char). Unfortunately this means that
-      // legitimate usage of signed chars is impossible, but this should be rare.
-      //
-      // Just return Boolean values as-is.
-      return number
-
+  private func wrapNumberIfNeeded(_ number: NSNumber) -> Any {
+    switch String(cString: number.objCType) {
+    case "q":
+      // "long long" might be larger than JS supports, so make it a string:
+      return WrappedNumber(type: .long, value: "\(number)").encoded
+    case "Q":
+      // "unsigned long long" might be larger than JS supports, so make it a string:
+      return WrappedNumber(type: .unsignedLong, value: "\(number)").encoded
     default:
-      // All documented codes should be handled above, so this shouldn"t happen.
-      throw Error.unknownNumberType(charValue: String(cType[0]), number: number)
```

Comment:
IIUC, this error was never thrown in practice because `objCType` returns a fixed set of possible strings that are all safe to encode.

Reply:
It is technically possible to provide something else here, but `NSNumber`’s [subclassing notes](https://developer.apple.com/documentation/foundation/nsnumber#Subclassing-Notes) explicitly advise against that. I guess if this is ever the case, it’s the client’s responsibility to prepare their custom object for encoding.